src/QsCompiler/Core/SymbolTable.fs

```diff
@@ -823,7 +829,31 @@ and NamespaceManager
                                 let msgArgs = [sprintf "%s.%s" epName.Namespace.Value epName.Name.Value; epSource.Value]
                                 (att.Offset, tId.Range |> orDefault |> QsCompilerDiagnostic.Error (ErrorCode.MultipleEntryPoints, msgArgs)) |> Seq.singleton |> returnInvalid 
                     | _ -> (att.Offset, tId.Range |> orDefault |> QsCompilerDiagnostic.Error (ErrorCode.InvalidEntryPointPlacement, [])) |> Seq.singleton |> returnInvalid
+                
+                // the attribute marks a unit test
+                elif tId.Namespace.Value = BuiltIn.TestOperation.Namespace.Value && tId.Name.Value = BuiltIn.TestOperation.Name.Value then
+                    let isUnitToUnit (signature : CallableSignature) = 
+                        let isUnitType = function 
+                            | Tuple _ | Missing -> false
+                            | Item (itemType : QsType) -> itemType.Type = UnitType
+                            | _ -> true // invalid type
+                        match signature.Argument.Items |> Seq.toList with 
+                        | [] -> signature.ReturnType |> isUnitType
+                        | [(_, argType)] -> argType |> isUnitType && signature.ReturnType |> isUnitType
+                        | _ -> false
+                    match box decl.Defined with 
+                    | :? CallableSignature as signature when signature |> isUnitToUnit && not (signature.TypeParameters.Any()) -> 
+                        let arg = att.Argument |> AttributeAnnotation.NonInterpolatedStringArgument (fun ex -> ex.Expression)
+                        match (if arg = null then null else arg.ToLowerInvariant()) with 
+                        | "quantumsimulator" | "tracesimulator" | "toffolisimulator" -> // error message below needs to be adapted when this changes!
```

Comment:
Seems like the comment here is a "ToDo". If it is, please add the "ToDo" to it to make it easier to find and track.